storage_backend/models/storage_backend.py

```diff
@@ -5,13 +5,38 @@
 # License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
 
 import base64
+import inspect
 import logging
 
-from odoo import fields, models
+from odoo import _, fields, models
 
 _logger = logging.getLogger(__name__)
 
 
+def _check_class_direct_func(instance, func_name):
```

Comment:
@guewen do you know how to work around this?
I need to check if the method is re-defined at component level but both approaches do not work.

Reply:
Couldn't the components implement a method with a different name? If it exists, use it, otherwise, use the generic one?

Reply:
yes but then I can also not define the method on the base adapter, that would be easier.
It's just that I'm trying to keep it consistent w/ the rest and understand what's going on.
To me, the class attached to the component is wrong or at least the module is wrong, no?

Reply:
I think that all components are registered in `odoo.addons.component.core` the same way all models are registered in `odoo.api`:

```
(Pdb) pp self.env["res.partner"].__class__
<class 'odoo.api.res.partner'>
(Pdb) pp self.env["sale.order"].__class__
<class 'odoo.api.sale.order'>
```

The thing is: these classes are like virtual classes which are built by aggregating all the classes found for a model/component. So they don't really have a proper module. We could have used the module of the first class found with `_name`, but I don't think we should handle differently than odoo, and it would maybe be even more confusing, because, for instance given the class FoorBar exist in module `odoo.addons.connector_foo`, the actual composite class `FooBar` built from all the classes found would not really be equal to the class `odoo.addons.connector_foo.FooBar`.

Reply:
ok, tnx. I've opted for the easy way: not define the method on the base class.